packages/client/src/components/form/FormFieldGenerator.tsx

```diff
@@ -869,11 +914,47 @@ class FormSectionComponent extends React.Component<Props> {
     })
   }
 
+  getOnClickHandle = (
+    fields: IFormField[],
+    field: IButtonFormField,
+    setFieldValue: (fieldName: string, value: any) => void,
+    setFieldError: (fieldName: string, message: string) => void
+  ) => {
+    const trigger = fields.find((f) => f.name === field.options.trigger)
+    if (trigger && isFieldHttp(trigger)) {
+      return () => {
+        const {
+          options: { url, ...requestOptions }
+        } = trigger
+        const fieldsDependentOnTrigger = getDependentFields(fields, trigger)
+        fetch(url, requestOptions)
+          .then((res) => res.text())
```

Comment:
This logic then again I would put in a new function like 

`function http(setFieldValue: fn)`
You can even make this a new component file in `packages/client/src/components/form/http.ts` or similar and the code would become

```ts
if (trigger && isFieldHttp(trigger)) {
  http(setFieldValue)
```

This is so that we can abstract all HTTP component related logic to a new module

Reply:
agreed, thanks @rikukissa. I am on it